test/test_logaggregator.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	c "github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-check"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/pkg/cluster"
+)
+
+type LogAggregatorSuite struct {
+	CLISuite
+}
+
+var _ = c.Suite(&LogAggregatorSuite{})
+
+func drain(ldrc chan *discoverd.Instance) {
+}
+
+func (s *LogAggregatorSuite) logAggregatorFailover(t *c.C) {
+	discd := s.discoverdClient(t)
+	srv := discd.Service("flynn-logaggregator")
+	leader, err := srv.Leader()
+	t.Assert(err, c.IsNil)
+
+	ldrc := make(chan *discoverd.Instance)
+	stream, err := srv.Leaders(ldrc)
+	t.Assert(err, c.IsNil)
+	defer stream.Close()
+
+	hostID, jobID, err := cluster.ParseJobID(leader.Meta["FLYNN_JOB_ID"])
+	t.Assert(err, c.IsNil)
+
+	for len(ldrc) > 0 {
+		<-ldrc
+	}
+
+	hc := s.hostClient(t, hostID)
+	t.Assert(hc.StopJob(jobID), c.IsNil)
+
+	select {
+	case <-ldrc:
+	case <-time.After(10 * time.Second):
+		t.Fatal("timeout wait for new logaggregator leader")
+	}
+}
+
+func (s *LogAggregatorSuite) TestLogReplication(t *c.C) {
+	app := s.newCliTestApp(t)
+
+	scale := app.flynn("scale", "sequential-printer=1")
+	app.waitFor(jobEvents{"sequential-printer": {"up": 1}})
+	t.Assert(scale, Succeeds)
+
+	for {
+		if app.flynn("log", "--raw-output").Output != "" {
+			break
```

Comment:
Is it worth waiting for a few lines of output before checking they are sequential?